framework/include/base/Assembly.h

```diff
@@ -768,87 +768,78 @@ class Assembly
   void copyNeighborShapes(MooseVariableField<T> & v);
   void copyNeighborShapes(unsigned int var);
 
+  /**
+   * Key structure for APIs manipulating global vectors/matrices
+   */
+  class GlobalDataKey
+  {
+    // Blessed classes
+    friend class Assembly;
+    friend class FEProblemBase;
+    friend class DisplacedProblem;
+    friend class ComputeMortarFunctor;
+    friend class NonlinearSystemBase;
+    GlobalDataKey() {}
+    GlobalDataKey(const GlobalDataKey &) {}
+  };
```

Comment:
what s the plan for directly modifying the petsc matrices if we re locking down all Assembly?

The tags help us do selective accumulations, do they also allow selective modifications?

Reply:
yes there are set/assignment operations